### ISSUE TITLE
Add >= 24 hard tasks requiring multi-hop graph traversal (hard >= 40% of pool)

### ISSUE THREAD

**Issue by davidrsch** [davidrsch/rrlmgraph-bench#52]
## Summary

Expand the benchmark task pool from 30 tasks to at least 54 tasks by adding ≥24 hard
tasks that require multi-hop graph traversal to answer correctly.

## Problem

The current pool of 30 tasks contains only 4 classified as difficulty `"hard"`. The
remaining 26 are `"easy"` or `"medium"`. This distribution has two consequences:

1. **Statistical power on the wrong sub-population.** If graph traversal is only
   advantageous on hard tasks (tasks that require looking up multiple related functions
   across files), then 30×5 = 150 total observations per strategy spread across 4 hard
   tasks gives ~20 observations per strategy per hard task. That is insufficient to
   resolve any effect within the hard sub-population.

2. **LLM prior dominates the result.** `no_context` (no retrieval at all) scores 0.698
   NDCG@5. The best retrieval strategy scores 0.788. All strategies compete for 0.090
   NDCG above what the LLM knows from pre-training. Most of this 0.090 headroom likely
   comes from the small number of hard tasks. Easy and medium tasks are largely solved
   by the LLM's pre-training regardless of retrieval quality.

Adding hard tasks is the primary way to increase headroom and test whether graph
traversal provides a genuine advantage over flat retrieval baselines.

## Definition of a hard task

A task qualifies as `"hard"` if answering it correctly requires at least one of:

- **Multi-hop lookup**: the answer involves a function that is not the seed function
  but is reached by following ≥2 edges from the entry point (e.g. A calls B, B calls C;
  the correct implementation must use C).
- **Cross-module dependency**: the relevant code is split across ≥2 source files with
  no obvious keyword overlap between the task description and any single file.
- **Co-change requirement**: the task requires modifying ≥2 functions that have a
  historical co-change relationship (i.e. they are frequently modified together).
- **Caller identification**: the task requires finding all callers of a function and
  updating each one (the BFS must traverse incoming edges, not just outgoing).

## Task structure requirements

Each new task must follow the existing task JSON schema. The `seed_node` field may
remain for `graph_rag_*` strategies but should also include
`"seed_node_required": false` to permit evaluation under the `rrlmgraph_agentic`
condition.

Example fields that distinguish a hard task:
```json
{
  "task_id": "task_031",
  "difficulty": "hard",
  "category": "cross_module",
  "description": "...",
  "seed_node": "pkg::entry_function",
  "seed_node_required": false,
  "ground_truth_nodes": ["pkg::entry_function", "pkg::helper_a", "other_module::dep_b"]
}
```

`ground_truth_nodes` must list every node the correct solution requires; NDCG is
computed against this set.

## Target distribution

After this issue is resolved:

| Difficulty | Current | Target |
|---|---|---|
| hard | 4 | ≥24 |
| medium | ~16 | ≥16 |
| easy | ~10 | ≥10 |
| **total** | **30** | **≥54** |

Hard tasks must be ≥40% of the total pool.

## Acceptance criteria

- [ ] ≥24 new task JSON files added to `inst/tasks/` (or relevant subfolder)
- [ ] All new tasks have `"difficulty": "hard"` and at least one of the multi-hop / cross-module / co-change / caller-identification qualifiers documented in the JSON
- [ ] All new tasks have verified `solution/` files and correct `ground_truth_nodes` arrays
- [ ] `run_full_benchmark(.dry_run = TRUE)` processes all new tasks without error
- [ ] `devtools::test()` passes with zero failures
- [ ] Hard tasks are ≥40% of pool after this change
